### ISSUE TITLE
Unhelpful exception when trying to parse illformed XML with ObjectReader

### ISSUE THREAD

**Issue by mikaelstaldal** [FasterXML/jackson-dataformat-xml#253]
(Using Jackson 2.8.9.)

I have an `ObjectReader`, created from an `XmlMapper`. When trying to use `readValue(String)`, and the supplied string is not well-formed XML (say `"foobar"`), I get a quite cryptic and unhelpful exception:

```
java.io.IOException: com.ctc.wstx.exc.WstxUnexpectedCharException: Unexpected character 'f' (code 102) in prolog; expected '<'
	at com.fasterxml.jackson.dataformat.xml.util.StaxUtil.throwXmlAsIOException(StaxUtil.java:24)
	at com.fasterxml.jackson.dataformat.xml.XmlFactory._createParser(XmlFactory.java:561)
	at com.fasterxml.jackson.dataformat.xml.XmlFactory.createParser(XmlFactory.java:433)
	at com.fasterxml.jackson.databind.ObjectReader.readValue(ObjectReader.java:1220)
	at 
	... 25 more
Caused by: com.ctc.wstx.exc.WstxUnexpectedCharException: Unexpected character 'f' (code 102) in prolog; expected '<'
 at [row,col {unknown-source}]: [1,1]
	at com.ctc.wstx.sr.StreamScanner.throwUnexpectedChar(StreamScanner.java:651)
	at com.ctc.wstx.sr.BasicStreamReader.nextFromProlog(BasicStreamReader.java:2126)
	at com.ctc.wstx.sr.BasicStreamReader.next(BasicStreamReader.java:1173)
	at com.fasterxml.jackson.dataformat.xml.XmlFactory._initializeXmlReader(XmlFactory.java:652)
	at com.fasterxml.jackson.dataformat.xml.XmlFactory._createParser(XmlFactory.java:559)
	... 28 more
```

It does not make any sense to throw a generic `IOException` when parsing from a String. I would expect to get `JsonProcessingException` (which I will get when parsing invalid JSON or YAML with Jackson).

**Comment by cowtowncoder** [comment]
I agree, it would make sense to use a higher-level exception; `IOException` should be reserved for I/O problems.

Could you create a simple test case to show the problem? I know it's quite trivial, but I have found out that often times my attempts differ from use cases of reporter and it saves time to use exact fail case (or close approximation).

**Comment by mikaelstaldal** [comment]
```
import com.fasterxml.jackson.databind.ObjectReader;
import com.fasterxml.jackson.dataformat.xml.XmlMapper;

import java.io.IOException;

public class JacksonTest {
    public static void main(String[] args) throws IOException {
        XmlMapper xmlMapper = new XmlMapper();
        ObjectReader objectReader = xmlMapper.reader();
        objectReader.readValue("foobar");
    }
}
```

**Comment by cowtowncoder** [comment]
Ah. So, I think this is dup of #236, and fix is in 2.9.0. I verified this locally.
It would be nice to backport this in 2.8.x, but change is bit bigger, and change while for better may cause problems for code that relies on catching existing (wrong) exception.
So I think fix is unfortunately 2.9 only.

Thank you for reporting this, providing repro.
